docs/user-guide/node-selection/node-selection.md

```diff
@@ -0,0 +1,200 @@
+  containers:
+  - name: nginx
+    image: nginx
+```
+
+Then add a nodeSelector like so:
+
+{% include code.html language="yaml" file="pod.yaml" ghlink="/docs/user-guide/node-selection/pod.yaml" %}
+
+When you then run `kubectl create -f pod.yaml`, the pod will get scheduled on the node that you attached the label to! You can verify that it worked by running `kubectl get pods -o wide` and looking at the "NODE" that the pod was assigned to.
+
+## Interlude: built-in node labels
+
+In addition to labels you [attach yourself](#step-one-attach-label-to-the-node), nodes come pre-populated
+with a standard set of labels. As of Kubernetes v1.4 these labels are
+
+* `kubernetes.io/hostname`
+* `failure-domain.beta.kubernetes.io/zone`
+* `failure-domain.beta.kubernetes.io/region`
+* `beta.kubernetes.io/instance-type`
+* `beta.kubernetes.io/os`
+* `beta.kubernetes.io/arch`
+
+## Alpha features: affinity and anti-affinity
+
+`nodeSelector` provides a very simple way to constrain pods to nodes with particular labels. The affinity/anti-affinity
+feature, currently in alpha, greatly expands the types of constraints you can express. The key enhancements are
+
+1. the language is more expressive (not just "AND of exact match")
+1. you can indicate that the rule is "soft"/"preference" rather than a hard requirement, so if the scheduler
+can't satisfy it, the pod will still be scheduled
+1. you can constrain against labels on other pods running on the node (or other topological domain),
+rather than against labels on the node itself, which allows rules about which pods can and cannot be co-located
+
+The affinity feature consists of two types of affinity, "node affinity" and "inter-pod affinity/anti-affinity."
+Node affinity is like the existing `nodeSelector` (but with the first two benefits listed above),
+while inter-pod affinity/anti-affinity constrains against pod labels rather than node labels, as
+described in the third item listed above, in addition to having the first and second properties listed above.
+
+`nodeSelector` continues to work as usual, but will eventually be deprecated, as node affinity can express
+everything that `nodeSelector` can express.
+
+### Node affinity (alpha feature)
+
+Node affinity was introduced in Kubernetes 1.2.
+Node affinity is conceptually similar to `nodeSelector` -- it allows you to constrain which nodes your
+pod is eligible to schedule on, based on labels on the node.
+
+There are currently two types of node affinity, called `requiredDuringSchedulingIgnoredDuringExecution` and
+`preferredDuringSchedulingIgnoredDuringExecution`. You can think of them as "hard" and "soft" respectively,
+in the sense that the former specifies rules that *must* be met for a pod to schedule onto a node (just like
+`nodeSelector` but using a more expressive syntax), while the latter specifies *preferences* that the scheduler
+will try to enforce but will not guarantee. The "IgnoredDuringExecution" part of the names means that, similar
```

Comment:
IgnoredDuringExecution is similar to nodeSelector functionality (functionality is probably not the right word). If labels...